api.planx.uk/modules/webhooks/service/updateTemplatedFlowEdits/index.ts

```diff
@@ -1,14 +1,77 @@
+import type { Node, NodeId } from "@opensystemslab/planx-core/types";
+import { gql } from "graphql-request";
+import * as jsondiffpatch from "jsondiffpatch";
+import isEmpty from "lodash/isEmpty.js";
+import { $api } from "../../../../client/index.js";
+import { getFlowData } from "../../../../helpers.js";
+import type { Flow } from "../../../../types.js";
+
 export const updateTemplatedFlowEdits = async (
   flowId: string,
   templatedFrom: string,
-  data: any,
+  data: Flow["data"],
 ) => {
-  // TODO
-  // 1. Fetch flows.data where id = templatedFrom
-  // 2. JSON diff against event `data`
-  // 3. Filter diff for nodes tagged 'customisation'
-  // 4. Transform diff to `templated_flow_edits` data shape
-  // 5. Update `templated_flow_edits`
-
-  return { flowId, templatedFrom, data };
+  // We fetch the live source template data, not its' latest published version, so it's a comparable diff to live incoming data
+  //   Eg we need to pick up if external portal pointer flowIds have changed
+  const sourceTemplate = await getFlowData(templatedFrom);
+  const sourceTemplateData = sourceTemplate?.data;
+  if (!sourceTemplateData) return;
+
+  // Because the templated flow only enables editing of nodes tagged "customisation" or their children,
+  //   the delta by default will only ever include nodes we need to record in `templated_flow_edits`
+  const delta = jsondiffpatch.diff(sourceTemplateData, data);
+  if (!delta) return;
+
+  // For each changed node property in the delta, extract its' current data state only
+  //   to store in templated_flow_edits (delta returns before and after state)
+  const templatedFlowEditsData: Record<NodeId, Partial<Node>> = {};
+  Object.entries(delta).forEach(([nodeId, nodeData]) => {
+    const updatedNodeData: Node["data"] = {};
+    const updatedKeys: string[] = nodeData?.data && Object.keys(nodeData.data);
+    updatedKeys?.forEach((updatedKey) => {
+      // TODO check how this handles removing properties eg 'description'
+      updatedNodeData[updatedKey] = data[nodeId]?.["data"]?.[updatedKey];
```

Comment:
Known TODO - let's come back to this one after we get more clarity and realistic examples of customisable nodes I think, unclear still how much of an issue/edge-case tracking deleted fields would actually be !